modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/template/test/TestTemplateContextContributor.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.notification.internal.template.test;
+
+import com.liferay.portal.kernel.template.TemplateContextContributor;
+
+import java.io.Serializable;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Richard Jeremias
+ */
+@Component(
+	property = "type=" + TemplateContextContributor.TYPE_GLOBAL,
+	service = TemplateContextContributor.class
+)
+public class TestTemplateContextContributor
+	implements TemplateContextContributor {
+
+	@Override
+	public void prepare(
+		Map<String, Object> contextObjects,
+		HttpServletRequest httpServletRequest) {
+
+		contextObjects.put(
+			"templateComponent", new InnerTemplateContextContributor());
+	}
+
+	@Component(property = "type=INNER", service = Serializable.class)
+	public static class InnerTemplateContextContributor
+		implements Serializable {
```

Comment:
@rijema it should be `TestSerializable`

Reply:
Sure. So, about the service, it would change from @Component(property = "type=INNER" to 	@Component(property = "type=SERIALIZABLE") or  @Component(property = "type=TEST")?

Reply:
To avoid errors like: > Task :apps:notification:notification-service:jar FAILED
error  : Malformed property 'INNER' on component 

This happened when I didn't insert the prefix type and this specific property, so it appears to be necessary to identify as a service.